### ISSUE TITLE
Documentation or code change to reuse user's existing caches for builds

### ISSUE THREAD

**Issue by binkley** [batect/batect#635]
I'm looking for either:

1. More clear documentation
2. Code changes

To reuse user's existing caches for builds.  A typical example is Maven or Gradle builds, which cache downloads in `~/.m2` or `~/.gradle`, respectively.  C/C++ builds have these concerns for distributed builds using `ccache` or similar tools.

As a local workaround, I have recommended to others that in a JVM (Java, Kotlin, et al) project root they create this symlink:
```
$ ln -s ~/.m2 .maven-cache
# Or with Gradle:
$ ln -s ~/.gradle .gradle-cache
```
And add the link to `.gitignore`.  Then in `batect.yml`:
```
containers:
  build-env:
    image: ... # Your image, such as adoptopenjdk:11-jdk-hotspot
    volumes:
      - local: .maven-cache
        container: /home/container-user/.m2
      # ... Other volume options, particularly "code" for the project root
```
It isn't clear from the documentation how this interacts with the Batect "cache" feature.

Goals:
- Do not download duplicate dependencies between `batect` builds and local builds directly using a tool (eg, `./mvnw`, `./gradlew`)
- Avoid duplicate disk space of caching downloads separately between local and `batect` builds.  Modern project can be disk hogs given the diverse range of dependencies
- Simple setup of Batect to manage these concerns

**Comment by charleskorn** [comment]
Hi @binkley, you should be able to use an absolute path or a path starting with `~` in the `local` path for a volume mount to do this. For example:

```yaml
containers:
  build-env:
    image: ... 
    volumes:
      - local: ~/.m2
        container: /home/container-user/.m2
```

This is just an ordinary Docker volume mount, and doesn't use a batect cache. You mentioned you looked at the documentation but how this works wasn't clear - so that I can improve the documentation, where were you looking for this information?

Having said that, I'd recommend you don't share these directories between containers and your host, for two reasons:

* some tools such as Gradle don't handle this well (see https://github.com/gradle/gradle/issues/851 for an example)
* some tools expect the directory to only be used by a single operating system (especially important when Windows and different line endings are involved)

**Comment by binkley** [comment]
This is such a better solution than the symlink -- thank you!

Yes, this was not obvious to me from the documentation.  Reading through the comments on the Gradle issue, this seems complex, and outside scope for Batect.  Personally, I'm not seeing problems, but the issue commenters do encounter them.  This is what I tried:
```
$ rm -rf ~/.m2  # or ~/.gradle
$ cd <project root with Batect setup, either Maven or Gradle>
$ ./batect build  # Massive download follows :)
```
No issues that I could see.  I checked the file perms on `~/.m2` and `~/.gradle`, and they were the same as if I had run `./mvnw` or `./gradlew` (respectively) from the project root.  Huzzah!

I can't speak to the issues you described (and thank you for pointing them out as potential pitfalls): None of the difficulties happened in my personal projects I tested with--maybe I got lucky.  Following up with GitHub Actions, CI seems good as well for Maven and Gradle.

Thank you for a better solution!
